CONTRIBUTING.md

```diff
@@ -1,100 +1,149 @@
+
+  ```
+  git pull upstream master
+  ```
+
+  **5.** Create a new branch.
 
-To update the Service Mesh Timeline, add the new service mesh in the [non-functional.js](https://github.com/layer5io/layer5/blob/master/src/collections/landscape/non-functional.js) file. Add `announce-date` key for the service mesh then add a `timeline-order` key having value 1+`maximum value of timeline-order` till now, to list the service mesh in the timeline correctly.
-#### <a name="blog">Adding a blog post</a>
+  ```
+  git checkout -b <your_branch_name>
+  ```
 
-1. In order to contribute a blog post, fork this repository, clone it, create a branch and navigate to the `src/collections/blog` directory.
-1. Create a copy of the [blog template](https://github.com/layer5io/layer5/tree/master/src/collections/blog/blog-template).
-1. Follow the instructions included in the blog template and name the new file after the title of the blog article.
-1. Entries will be listed in chronological order automatically.
+  **6.** Install the dependencies for running the site.
 
-#### <a name="news">Adding News</a>
+  ```
+  make setup-libs
+  ```
 
-1. In order to add/update news items, fork this repository, clone it, create a branch and navigate to the `src/collections/news` directory.
-1. Create a copy of the [news template](https://github.com/layer5io/layer5/tree/master/src/collections/news/0000-00-00-news-template).
-1. Follow the instructions included in the news template and name the new file after the title of the news article.
-1. Entries will be listed in chronological order automatically.
+  **7.** Make the desired changes.
 
-#### <a name="events">Adding an Event</a> [create new issue](https://github.com/layer5io/layer5/issues/new?assignees=&labels=area%2Fevents&template=events.md&title=%5BEvent%5D)
+  **8.** Run the site locally to preview changes.
 
+  ```
+  make site
+  ```
 
-## Contributor Instructions
+This will run a local web server with "live reload" conveniently enabled. ( **NOTE**: while using make command on Windows, there sometimes arises an error in identifying the command even after it is installed (unrecognized command), this is because the PATH for the binary might not be set correctly ).
 
-Before building the site, ensure that you have the following dependencies installed on your system.
+  **9.** Track your changes.
 
-- [Nodejs and npm](https://nodejs.org/en/): To verify run `node -v` & `npm -v`
+  ```
+  git add .
+  ```
 
-### Building the site
+  **10.** Commit your changes. To contribute to this project, you must agree to the [Developer Certificate of Origin (DCO)](## Signing-off on Commits (Developer Certificate of Origin)) for each commit you make.
 
-Run the following:
+  ```
+  git commit --signoff -m "<commit subject>"
+  ```
 
-1. `make setup-libs`-To install the dependencies for the running the site.
-1. `make site`-To run the website locally.
 
```

Comment:
maybe before pushing the changes, we could ask them to pull changes in the master from upstream and rebase their branch with the master branch. Something like in the below images : 

![](https://imgur.com/SlKN5vY.png)

![](https://imgur.com/2U1VvhP.png)